service/forward_service.hh

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2021-present ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <seastar/core/distributed.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/sharded.hh>
+
+#include "database_fwd.hh"
+#include "locator/token_metadata.hh"
+#include "message/messaging_service_fwd.hh"
+#include "query-request.hh"
+
+namespace service {
+
+// forward_service is a sharded service responsible for distributing and
+// executing aggregation requests across a cluster.
```

Comment:
It will be nice to explain the overall logic and give an example of how a query is split.